### PR TITLE
Volume Group Resource CRD operations

### PR DESCRIPTION
**Pull request by prajwalpatil25**
# Description
Implemented Create, Read and Delete operations of volume group resource and added acceptance tests.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [X] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [X] I have verified that new and existing unit tests pass locally with my changes
- [X] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Acceptance tests
image (1)

### CODE REVIEW

File: powerstore/resource_volume_group.go [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+				Optional:            true,
+				Computed:            true,
+				Description:         "Description for the volume group.",
+				MarkdownDescription: "Description for the volume group.",
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(1),
+				},
+			},
+
+			"volume_ids": schema.SetAttribute{
```
Comment (shenda1) [inline powerstore/resource_volume_group.go:69]: Please check if the volume names needs to be supported

File: powerstore/resource_volume_group.go [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+			},
+
+			"is_write_order_consistent": schema.BoolAttribute{
+				Optional:            true,
+				Computed:            true,
+				Description:         "Determines whether snapshot sets of the group will be write order consistent.",
+				MarkdownDescription: "Determines whether snapshot sets of the group will be write order consistent.",
+			},
+
+			"protection_policy_id": schema.StringAttribute{
```
Comment (shenda1) [inline powerstore/resource_volume_group.go:89]: Please check if protection policy name needs to be supported

File: powerstore/resource_volume_group.go [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	//Get Volume Group ID from state
+	volumeGroupID := state.ID.ValueString()
+
+	//Delete Volume Group by calling API
+	_, err := r.client.PStoreClient.DeleteVolumeGroup(context.Background(), volumeGroupID)
```
Comment (shenda1) [inline powerstore/resource_volume_group.go:192]: Do we need to delete volumes associated with the volume group before deleting the volume group?

File: powerstore/resource_volume_group.go (hunk 1 above)
Comment (prajwalpatil25) [reply powerstore/resource_volume_group.go:69]: yes, volume names and protection policy names needs to be supported. Will have different PR to address these later.

File: powerstore/resource_volume_group.go (hunk 3 above)
Comment (prajwalpatil25) [reply powerstore/resource_volume_group.go:192]: As discussed, volumes would not be removed from volume group while deleting it. An error would be thrown to user and user needs to address it manually. Same applies to the protection policy attached to the volume group.

**Review by shenda1 — DISMISSED** [review]

**Review by Sakshi-dell — APPROVED** [review]

**Review by shenda1 — APPROVED** [review]

**Pull request merged**